pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/generic/AvroRecordBuilderImpl.java

```diff
@@ -45,7 +47,12 @@
      */
     @Override
     public GenericRecordBuilder set(String fieldName, Object value) {
-        avroRecordBuilder.set(fieldName, value);
+        if (value instanceof GenericRecord) {
```

Comment:
I think the logic here is a bit wrong. I think you can just retrieve the avro record.

```
if (value of instanceof GenericRecord) {
    if (value of instanceof GenericAvroRecord) {
        avroRecordBuilder.set(fieldName, ((GenericAvroRecord)value).getAvroRecord());
    } else {
        throw new IllegalArgumentException("Avro Record Builder doesn't support non-avro record as a field");
    }
} else {
    avroRecordBuilder.set(fieldName, value);
}
```